### DISCUSSION TITLE
Hello I need your help

### DISCUSSION THREAD

**Discussion by alexmathew11** [zepp-health/zeppos-forum#67] [Q&A]
Hello basically I am trying to make an app if we click a button it wont go to page it will show another button without switching page please give me the code

**Comment by zdunex25** [comment]
https://docs.zepp.com/docs/reference/device-app-api/newAPI/ui/setProperty/#propertyid

**Comment by alexmathew11** [comment]
I guess you didn't get it I want for Amazfit band 7 and it will be 1.0

**Comment by zdunex25** [comment]
You did not mention anything about AB7 in you question, so your answer is: you cannot.

**Comment by alexmathew11** [comment]
You cannot just say no every thing is possible accept by defeating death

**Comment by zdunex25** [comment]
Go ahead and hack your way over system limitations.

**Comment by MyoungXUE** [comment]
Hello, you can refer to this code.

```js
Page({
  data: {
    visible: false
  },
  build() {
    const visibleButton = hmUI.createWidget(hmUI.widget.BUTTON, {
      x: (480 - 400) / 2,
      y: 300,
      w: 400,
      h: 100,
      radius: 12,
      normal_color: 0xfc6950,
      press_color: 0xfeb4a8,
      text: 'Visible',
      click_func: (button_widget) => {
        console.log('toggle')
      },
    })
    visibleButton.setProperty(hmUI.prop.VISIBLE, this.data.visible)

    const toggleButton = hmUI.createWidget(hmUI.widget.BUTTON, {
      x: (480 - 400) / 2,
      y: 120,
      w: 400,
      h: 100,
      radius: 12,
      normal_color: 0xfc6950,
      press_color: 0xfeb4a8,
      text: 'Toggle',
      click_func: (button_widget) => {
        this.data.visible = !this.data.visible
        visibleButton.setProperty(hmUI.prop.VISIBLE, this.data.visible)
      }
    })
  }
})
```

**Reply by zdunex25** [reply]
According to documentation **setProperty** does not work on zeppos 1.0 and amazfit band 7 runs zeppos 1.0, so how could it work?

**Reply by MyoungXUE** [reply]
> According to documentation **setProperty** does not work on zeppos 1.0 and amazfit band 7 runs zeppos 1.0, so how could it work?

Hello, it may be that the document is not clearly expressed. Can you show me a screenshot of where the document is marked, so we can improve the description?

Back to the question, this code is verified to work in band7 on real machine, you can verify it with similar implementation in simulator or real machine.

**Reply by alexmathew11** [reply]
Hello can you please simplify the code like


const visibleButton = hmUI.createWidget(hmUI.widget.BUTTON, {
      x: 20,
      y: 300,
      w: 100,
      h: 100,
      radius: 12,
      normal_color: 0xfc6950,
      press_color: 0xfeb4a8,
      text: 'Visible',
      click_func: (button_widget) => {
    const toggleButton = hmUI.createWidget(hmUI.widget.BUTTON, {
      x: 20,
      y: 120,
      w: 400,
      h: 100,
      radius: 12,
      normal_color: 0xfc6950,
      press_color: 0xfeb4a8,
      text: 'Toggle',
      click_func: (button_widget) => {
        visibleButton.setProperty(hmUI.prop.VISIBLE, this.data.visible)
      }
    })

I know it won't gonna work but I won't it very simple. I don't know if you will understand or not because I am just a new starter

**Reply by MyoungXUE** [reply]
@alexmathew11 Sorry, I didn't understand what you meant.

I'll add some textual explanations to the code I provided.

This is a sample code that shows how to control the display and hide of `visibleButton` by clicking `toggleButton `, the position and width of the two buttons need to be adjusted manually by you to fit the band 7 screen.

First we create `visibleButton` and set its `VISIBLE` property to `false`, then we create `toggleButton` and use the `setProperty` API to modify the `VISIBLE` property in its `click_func` to control the display of the ` toggleButton` to show and hide.

Hope this helps you!

**Reply by zdunex25** [reply]
> Hello, it may be that the document is not clearly expressed. Can you show me a screenshot of where the document is marked, so we can improve the description?

Here you go:
![image](https://user-images.githubusercontent.com/1674725/217189788-e23a9f59-28ac-48e3-8bf0-448ff4586866.png)

**Reply by MyoungXUE** [reply]
> > Hello, it may be that the document is not clearly expressed. Can you show me a screenshot of where the document is marked, so we can improve the description?
> 
> Here you go: ![image](https://user-images.githubusercontent.com/1674725/217189788-e23a9f59-28ac-48e3-8bf0-448ff4586866.png)

I see what you mean, the documentation version is really not good enough, this piece we will improve as soon as possible.

For the current discussion, this API is also available in Zepp OS 1.0, you can refer to https://docs.zepp.com/docs/1.0/reference/device-app-api/hmUI/setProperty/
